docs/running_psalm/plugins/plugins_type_system.md

```diff
@@ -41,16 +41,40 @@ The classes are as follows:
 
 `TResource` - denotes the `resource` type (e.g. a file handle).
 
+`TClosedResource` - denotes the `resource` type that has been closed (e.g. a file handle through `fclose()`).
+
+`TAssertionFalsy` - Represents any value reduced to false when computed in boolean context. This is used for assertions
+
+`TConditional` - Internal representation of a conditional return type in phpdoc. For example ($param1 is int ? int : string)
+
+`TIntMask` - Represents the type that is the result of a bitmask combination of its parameters. `int-mask<1, 2, 4>` corresponds to `1|2|3|4|5|6|7`
+
+`TIntMaskOf` - as above, but used with with a reference to constants in code`int-mask<MyClass::CLASS_CONSTANT_*>` will corresponds to `1|2|3|4|5|6|7` if there are three constant 1, 2 and 4
+
+`TKeyOfClassConstant` - Represents an offset of a class constant array.
+
+`TTemplateIndexedAccess` - To be documented
+
+`TTemplateKeyOf` - To be documented
+
+`TTypeAlias` - To be documented
+
+`TValueOfClassConstant` - To be documented
```

Comment:
Same as `TKeyOfClassConstant`, but for values, I suppose.

Reply:
Added, thanks